### PR TITLE
Add com.github.varnholt.dynablaster_revenge

### PR DESCRIPTION
**Pull request by pizzadude**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. I submitted an issue here a couple months ago, but no reply. [https://github.com/varnholt/dynablaster_revenge/issues/16](https://github.com/varnholt/dynablaster_revenge/issues/16)
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

The only patch I included is to change the game settings file location, so that saving settings works in the flatpak. I didn't submit it upstream because upstream hasn't replied to my original issue yet in months.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.varnholt.dynablaster_revenge.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<application>
+  <id type="desktop">com.github.varnholt.dynablaster_revenge</id>
+  <project_license>CC BY-NC 4.0</project_license>
+  <metadata_license>CC BY-NC 4.0</metadata_license>
```
Comment (hfiguiere) [inline com.github.varnholt.dynablaster_revenge.appdata.xml:5]: this is not an acceptable meta data license. (it's THIS file, just to be clear)

File: com.github.varnholt.dynablaster_revenge.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  <provides>
+    <id>com.github.varnholt.dynablaster_revenge.desktop</id>
+  </provides>
+  <name>Dynablaster Revenge</name>
+  <summary></summary>
```
Comment (hfiguiere) [inline com.github.varnholt.dynablaster_revenge.appdata.xml:10]: summary must not be empty.

File: com.github.varnholt.dynablaster_revenge.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  </description>
+  <url type="homepage">https://github.com/varnholt/dynablaster_revenge</url>
+  <screenshots>
+    <screenshot type="default">https://raw.githubusercontent.com/varnholt/dynablaster_revenge/master/doc/screenshot_2.jpg</screenshot>
+    <screenshot>https://raw.githubusercontent.com/varnholt/dynablaster_revenge/master/doc/screenshot_3.jpg</screenshot>
+  </screenshots>
+  <content_rating type="oars-1.1">
+    <content_attribute id="violence-cartoon">moderate</content_attribute>
+    <content_attribute id="violence-fantasy">moderate</content_attribute>
+    <content_attribute id="violence-realistic">none</content_attribute>
```
Comment (hfiguiere) [inline com.github.varnholt.dynablaster_revenge.appdata.xml:27]: all the one set to `none` can be omitted.

File: dynablaster.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+cd /app/bin/
+/app/bin/dynablaster
```
Comment (hfiguiere) [inline dynablaster.sh:4]:
```suggestion
exec /app/bin/dynablaster
```

File: savefix.patch [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+--- a/client/src/game/gamesettings.cpp
++++ b/client/src/game/gamesettings.cpp
+@@ -1,8 +1,11 @@
+ #include "gamesettings.h"
++#include <QStandardPaths>
+ 
+ #include <cstdint>
+ 
+-#define SETTINGS_FILE "data/game.ini"
++QString settingsPath = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/dynablaster/game.ini";
```
Comment (hfiguiere) [inline savefix.patch:12]:
```suggestion
+const QString settingsPath = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/dynablaster/game.ini";
```

File: com.github.varnholt.dynablaster_revenge.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        "sources": [
+            {
+                "type": "git",
+                "url": "https://github.com/varnholt/dynablaster_revenge.git",
+                "commit": "08fb11fb2d5dd94e1ea0e447d95cccb375dfc7e5"
+            },
+            {   "type": "patch",
+                "path": "savefix.patch"
+            },
+
```
Comment (hfiguiere) [inline com.github.varnholt.dynablaster_revenge.json:32]:
```suggestion
```

File: com.github.varnholt.dynablaster_revenge.json [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        "name": "dynablaster",
+        "buildsystem": "qmake",
+        "no-make-install": true,
+        "subdir": "client",
+        "sources": [
+            {
+                "type": "git",
+                "url": "https://github.com/varnholt/dynablaster_revenge.git",
+                "commit": "08fb11fb2d5dd94e1ea0e447d95cccb375dfc7e5"
```
Comment (hfiguiere) [inline com.github.varnholt.dynablaster_revenge.json:27]: any reason to not use the tagged release?

File: com.github.varnholt.dynablaster_revenge.json [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=pulseaudio"
+    ],
+    "command": "dynablaster.sh",
+    "cleanup": [
+        "Makefile"
+    ],
+    "modules": ["shared-modules/glu/glu-9.json",
```
Comment (hfiguiere) [inline com.github.varnholt.dynablaster_revenge.json:17]:
```suggestion
    "modules": [
        "shared-modules/glu/glu-9.json",
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]
have the .destkop and appdata file been submitted upstream?

**Comment by hfiguiere** [comment]
bot, build com.github.varnholt.dynablaster_revenge

**Comment by pizzadude** [comment]
> have the .destkop and appdata file been submitted upstream?

No, I didn't because the developer is unresponsive.

**Comment by hfiguiere** [comment]
> > have the .destkop and appdata file been submitted upstream?
> 
> No, I didn't because the developer is unresponsive.

If you don't submit them you clear won't get a reply ever. That's for sure.

**Comment by pizzadude** [comment]
> > > have the .destkop and appdata file been submitted upstream?
> > 
> > 
> > No, I didn't because the developer is unresponsive.
> 
> If you don't submit them you clear won't get a reply ever. That's for sure.

What folder in the upstream repo should I submit them to?
https://github.com/varnholt/dynablaster_revenge

File: com.github.varnholt.dynablaster_revenge.json (hunk 7 above)
Comment (pizzadude) [reply com.github.varnholt.dynablaster_revenge.json:27]: There were fixes to the project since then, and the project doesn't update often anyways.

**Comment by pizzadude** [comment]
bot, build com.github.varnholt.dynablaster_revenge

**Comment by pizzadude** [comment]
bot, build com.github.varnholt.dynablaster_revenge

**Comment by pizzadude** [comment]
bot, build com.github.varnholt.dynablaster_revenge

**Comment by pizzadude** [comment]
bot, build com.github.varnholt.dynablaster_revenge

File: com.github.varnholt.dynablaster_revenge.json [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    "command": "dynablaster.sh",
+    "cleanup": [
+        "Makefile"
+    ],
+    "modules": [
+        "shared-modules/glu/glu-9.json",
+    {
+        "name": "dynablaster",
+        "only-arches": [
+            "x86_64"
```
Comment (hfiguiere) [inline com.github.varnholt.dynablaster_revenge.json:22]: this doesn't do what you think it does...

File: com.github.varnholt.dynablaster_revenge.json (hunk 7 above)
Comment (hfiguiere) [reply com.github.varnholt.dynablaster_revenge.json:31]: what are these fixes?

Comment (pizzadude) [reply com.github.varnholt.dynablaster_revenge.json:31]: Memory leak fixes, some build fixes

**Comment by pizzadude** [comment]
bot, build com.github.varnholt.dynablaster_revenge

**Comment by pizzadude** [comment]
It works.

**Pull request closed without merging**